src/imgui/ImGuiMachine.cc

```diff
@@ -142,6 +156,46 @@ void ImGuiMachine::paintSelectMachine(const MSXMotherBoard* motherBoard)
 			ImGui::Separator();
 		}
 
+		auto showMachine = [&](MachineInfo& info, bool doubleClickToSelect) {
+			bool ok = getTestResult(info).empty();
+			im::StyleColor(!ok, ImGuiCol_Text, getColor(imColor::ERROR), [&]{
+				bool selected = info.configName == newMachineConfig;
+				if (ImGui::Selectable(info.displayName.c_str(), selected,
+						doubleClickToSelect ? ImGuiSelectableFlags_AllowDoubleClick: ImGuiSelectableFlags_None)) {
+					newMachineConfig = info.configName;
+					if (ok && (doubleClickToSelect ? ImGui::IsMouseDoubleClicked(ImGuiMouseButton_Left) : true)) {
+						showSelectMachine = false; // close window
+						manager.executeDelayed(makeTclList("machine", newMachineConfig));
+						addRecentItem(recentMachines, newMachineConfig);
+					}
+				}
+				if (selected) {
+					if (ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+				}
+				if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+					im::ItemTooltip([&]{
+						printConfigInfo(info);
+					});
+				}
+			});
+		};
+
+		im::TreeNode("Recently used", recentMachines.empty() ? ImGuiTreeNodeFlags_None : ImGuiTreeNodeFlags_DefaultOpen, [&]{
+			if (recentMachines.empty()) {
+				ImGui::TextUnformatted("(none)"sv);
```

Comment:
In an earlier version, when `recentMachines` was empty, you wouldn't show this section at all. Now you show `(none)`. But are fine I think.

Maybe I have a slight personal preference for not showing the section. Normally this only happens once (unless you delete your `imgui.ini` file). And maybe not showing a not-yet-useful section for new users is better?

Reply:
Another point of view: it can show that we have a certain feature (as a teaser), which will quickly become useful. 

And it takes only one line in the window, as I do make the treenode closed if there are no recent machines.